docs/proposals/scheduling/20221227-cpu-static-reservation.md

```diff
@@ -0,0 +1,160 @@
+---
+title: cpu static reservation
+authors:
+  - "@lucming"
+reviewers:
+  - "@eahydra"
+  - "@hormes"
+  - "@yihuifeng"
+  - "@honpey"
+  - "@zwzhang0107"
+  - "@jasonliu747"
+creation-date: 2022-12-27
+last-updated: 2022-12-27
+---
+
+# CPU static Reservation
+
+## Table of Contents
+
+<!--ts-->
+
+- [CPU static Reservation](#cpu-static-reservation)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+    - [Implementation Details](#implementation-details)
+      - [koordlet](#koordlet)
+      - [koord-manager](#koord-manager)
+      - [koord-scheduler](#koord-scheduler)
+      - [example](#example)
+  - [Implementation History](#implementation-history)
+<!--te-->
+
+## Summary
+The proposal provides a scheduling mechanism to statically reserve resources on nodes, such as reserving 0-6 cores of cpu on the node.
+
+## Motivation
+
+
+### Goals
+
+- statically reserving cpu resources on node
+
+### Non-Goals/Future Work
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+As a k8s cluster administrator, there are not only containers running on the node of the k8s cluster, there may be other processes as well, in order to ensure the isolation of the containers and the original processes.
+At the cpu level, we usually want normal processes and containers to run on different cpu cores to reduce/avoid the impact of their resource usage.   
+eg：   
+totoal cpu on node:24;   
+expected performance:   
+normal process use:0-11;    
+container use:12-23 
+
+#### Story 2
+
+
+### Implementation Details
+
+#### koordlet   
+1. When koordlet report the `NodeResourceTopology` of the node, it will ignores the reserved cpu cores on node annotation. and will combine the reserved cpus of kubelet and `node.annotations`.   
+eg:  
+kubelet.args["reserved-cpus"]="6-7";  
+node.annotation["koordinator.sh/cpus-reserved"]="0-6"  
```

Comment:
It is recommended to add a separate chapter describing the annotation.
In addition, this annotation can not only reserve CPU, but also support other resources. In addition, the reserved CPU can be of LSE type by default, but it should also be considered that the reserved CPU is LSR.